cleanup_utils/aws_cleanup.py

```diff
@@ -0,0 +1,227 @@
+
+DEFAULT_POLICY_NAME = "allow_access_s3_bucket"
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger('integration_cleanup')
+
+
+# Classes #####################################################################
+
+class AwsCleanupInstance:
+    """
+    Handles the cleanup of IAM users, Policies and Buckets related to CircleCI
+    """
+    def __init__(self, age_limit, aws_access_key_id, aws_secret_access_key, dry_run=False):
+        """
+        Set's up AWS connections and clients
+        """
+        self.age_limit = age_limit
+        self.dry_run = dry_run
+        self.cleaned_up_hashes = []
+
+        self.session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        self.iam_client = self.session.client('iam')
+        self.s3_client = self.session.client('s3')
+        self.s3_resource = self.session.resource('s3')
+
+    def delete_bucket(self, bucket_name):
+        """
+        Deletes a S3 bucket and all of it's files
+        """
+        if self.dry_run:
+            return
+        try:
+            logger.info("Deleting bucket %s.", bucket_name)
+            bucket = self.s3_resource.Bucket(bucket_name)
+            bucket.object_versions.delete()
+            bucket.objects.all().delete()
+            bucket.delete()
+        except self.s3_client.exceptions.NoSuchBucket:
+            # Ignore if the bucket doesn't exist
+            logger.warning("The bucket %s did not exist, skipping.", bucket_name)
+        except botocore.exceptions.ClientError as e:
+            # Ignore if the bucket doesn't exist
+            logger.error(
+                "Untreated error when deleting %s: %s",
+                bucket_name,
+                e.response['Error']['Message']
+            )
+
+    def delete_user_policy(self, username, policy_name):
+        """
+        Deletes policy associated to user
+        """
+        if self.dry_run:
+            return
+        self.iam_client.delete_user_policy(
+            UserName=username,
+            PolicyName=policy_name
+        )
+
+    def delete_user_access_key(self, username, access_key):
+        """
+        Deletes a user's access key
+        """
+        if self.dry_run:
+            return
+        self.iam_client.delete_access_key(
+            UserName=username,
+            AccessKeyId=access_key['AccessKeyId']
+        )
+
+    def delete_user(self, username):
+        """
+        Deletes a IAM user
+        """
+        if self.dry_run:
+            return
+        self.iam_client.delete_user(UserName=username)
+
+    def get_iam_users(self):
+        """
+        Lists all IAM users
+        """
+        users = []
+        paginator = self.iam_client.get_paginator('list_users')
+        for page in paginator.paginate():
+            for user in page['Users']:
+                users.append(user)
```

Comment:
@giovannicimolin, unless `page['Users']` is a generator, can't you just use `users.extend(page['Users'])`?

Reply:
Similar changes can be done for all the pagination code.